packages/db/scripts/migrations/migrate-quizzes-to-v2.ts

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env tsx
+
+/**
+ * Quiz V1 to V2 Migration Script
+ *
+ * Purpose: Migrate legacy quiz format (V1) to new discriminated union format (V2)
+ *
+ * V1 Format: Array of multiple-choice questions
+ * V2 Format: Object with version field and questions array supporting multiple types
+ *
+ * Safety: Run with --dry-run first to preview changes
+ */
+import { upgradeQuizzes } from "../../../aila/src/protocol/schemas/quiz/conversion/lessonPlanQuizMigrator";
+import { aiLogger } from "@oakai/logger";
+
+import { PrismaClient } from "@prisma/client";
+
+const log = aiLogger("db");
+
+function hasQuizzes(output: unknown): boolean {
+  if (!output || typeof output !== "object") return false;
+
+  const data = output as {
+    lessonPlan?: { starterQuiz?: unknown; exitQuiz?: unknown };
+  };
+  return Boolean(data.lessonPlan?.starterQuiz || data.lessonPlan?.exitQuiz);
+}
+
+async function main() {
+  const isDryRun = process.argv.includes("--dry-run");
+  const includeDeleted = process.argv.includes("--include-deleted");
+  const showHelp = process.argv.includes("--help");
+
+  if (showHelp) {
+    log.info(`
+Quiz V1 to V2 Migration
+
+Usage: pnpm tsx migrate-quizzes-to-v2.ts [--dry-run] [--include-deleted]
+
+Options:
+  --dry-run          Preview changes without modifying database
+  --include-deleted  Also migrate soft-deleted sessions
+  --help             Show this help
+`);
+    process.exit(0);
+  }
+
+  const prisma = new PrismaClient();
+
+  log.info(`Starting quiz migration (${isDryRun ? "DRY RUN" : "LIVE"})`);
+  if (includeDeleted) {
+    log.info("Including deleted sessions");
+  }
+
+  try {
+    // Find all sessions with quizzes
+    // Note: We'll check for quiz presence in the code since Prisma JSON filtering is limited
+    const allSessions = await prisma.appSession.findMany({
+      where: includeDeleted ? {} : { deletedAt: null },
+      select: { id: true, output: true },
+    });
+
+    // Filter sessions that actually have quizzes
+    const sessionsWithQuizzes = allSessions.filter((session) =>
+      hasQuizzes(session.output),
+    );
+
+    log.info(
+      `Found ${allSessions.length} total sessions, ${sessionsWithQuizzes.length} have quizzes`,
+    );
+
+    let upgraded = 0;
```

Comment:
I was about to leave a comment about how I prefer the term 'update' to 'upgrade' because the latter feels qualitative and subjective.

Thought I'd look it up and have now learned that it's commonly accepted that major version changes are considered 'upgrades' and minor/path versions are 'updates'. Can't say it's my fave but I guess it makes sense!